File: GameLibrary/src/cc/lib/zombicide/anims/AscendingAngelDeathAnimation.java
package cc.lib.zombicide.anims;

import cc.lib.game.AGraphics;
import cc.lib.game.GRectangle;
import cc.lib.zombicide.ZCharacter;

public class AscendingAngelDeathAnimation extends DeathAnimation {

    public AscendingAngelDeathAnimation(ZCharacter a) {
        super(a);
        setDuration(4000);
    }

    @Override
    protected void draw(AGraphics g, float position, float dt) {
        super.draw(g, position, dt);
        GRectangle rect = new GRectangle(actor.getRect());
        rect.y -= rect.h * 3 * position;
        g.setTransparencyFilter(.5f - position/3);
        g.drawImage(actor.getImageId(), rect);
        g.removeFilter();
    }

    @Override
    protected void onDone() {
        super.onDone();
        ((ZCharacter)actor).setFallen(true);
    }
}

File: Zombicide/src/cc/lib/zombicide/anims/AscendingAngelDeathAnimation.kt
package cc.lib.zombicide.anims

import cc.lib.game.AGraphics
import cc.lib.game.GRectangle
import cc.lib.zombicide.ZCharacter

class AscendingAngelDeathAnimation(a: ZCharacter) : DeathAnimation(a) {
    override fun draw(g: AGraphics, position: Float, dt: Float) {
        super.draw(g, position, dt)
        val rect = GRectangle(actor.getRect())
        rect.y -= rect.h * 3 * position
        g.setTransparencyFilter(.5f - position / 3)
        g.drawImage(actor.imageId, rect)
        g.removeFilter()
    }

	override fun hidesActor(): Boolean {
		return true
	}

    override fun onDone() {
        super.onDone()
        (actor as ZCharacter).setFallen(true)
    }

    init {
        duration = 4000
    }
}